BlueprintUICommonControls/Sources/AttributedLabel.swift

```diff
@@ -96,27 +96,46 @@ public struct AttributedLabel: Element, Hashable {
         configure(&self)
     }
 
-    public var content: ElementContent {
-        struct Measurer {
-            private static let prototypeLabel = LabelView()
+    // MARK: Element
 
-            func measure(model: AttributedLabel, in constraint: SizeConstraint, environment: Environment) -> CGSize {
-                let label = Self.prototypeLabel
-                label.update(model: model, environment: environment, isMeasuring: true)
-                return label.sizeThatFits(constraint.maximum)
-            }
+    /// The text to pass to the underlying `UILabel`, normalized for display if necessary.
+    var displayableAttributedText: NSAttributedString {
+        if needsTextNormalization || linkDetectionTypes.isEmpty == false {
+            return attributedText.normalizingForView(with: numberOfLines)
+        } else {
+            return attributedText
         }
+    }
+
+    @_spi(BlueprintAttributedLabel)
+    /// Set this if you can guarantee that your label implementation will not need string normalization.
+    /// For example, a custom label type wrapping `AttributedLabel` may set this to `false`.
+    /// You can check if this value should be false via `NSAttributedString.needsNormalizingForView(...)`
+    public var needsTextNormalization: Bool = true
+
+    private static let prototypeLabel = LabelView()
+
+    public var content: ElementContent {
 
-        return ElementContent { sizeConstraint, environment -> CGSize in
-            Measurer().measure(model: self, in: sizeConstraint, environment: environment)
+        // We create this outside of the measurement block so it's called fewer times.
+        let text = displayableAttributedText
+
+        return ElementContent { constraint, environment -> CGSize in
+            let label = Self.prototypeLabel
+            label.update(model: self, text: text, environment: environment, isMeasuring: true)
+            return label.sizeThatFits(constraint.maximum)
         }
     }
 
     public func backingViewDescription(with context: ViewDescriptionContext) -> ViewDescription? {
-        LabelView.describe { config in
+
+        // We create this outside of the application block so it's called fewer times.
+        let text = displayableAttributedText
+
+        return LabelView.describe { config in
             config.frameRoundingBehavior = .prioritizeSize
             config.apply { view in
-                view.update(model: self, environment: context.environment, isMeasuring: false)
+                view.update(model: self, text: text, environment: context.environment, isMeasuring: false)
```

Comment:
🐴 

I _think_ this builds the case for the storage approach. It would remove a parameter here and perhaps make juggling this concern easier to manage.